src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -442,4 +498,95 @@ private Model GetVocabularyDictionary()
             }
         }
     }
+
+    public sealed class WordEmbeddingsEstimator : TrivialEstimator<WordEmbeddingsTransform>
+    {
+        public WordEmbeddingsEstimator(IHostEnvironment env, string inputColumn, string outputColumn,
+           WordEmbeddingsTransform.PretrainedModelKind? modelKind = WordEmbeddingsTransform.PretrainedModelKind.Sswe)
+            : this(env, new WordEmbeddingsTransform(env, inputColumn, outputColumn, modelKind))
+        {
+        }
+
+        public WordEmbeddingsEstimator(IHostEnvironment env,
+            WordEmbeddingsTransform.PretrainedModelKind? modelKind = WordEmbeddingsTransform.PretrainedModelKind.Sswe,
+            string customLookupTable = null, params WordEmbeddingsTransform.Column[] columns)
+            : this(env, new WordEmbeddingsTransform(env, modelKind, customLookupTable, columns))
+        {
+        }
+
+        public WordEmbeddingsEstimator(IHostEnvironment env, WordEmbeddingsTransform transformer)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(WordEmbeddingsEstimator)), transformer)
+        {
+        }
+
+        public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            Host.CheckValue(inputSchema, nameof(inputSchema));
+            var result = inputSchema.Columns.ToDictionary(x => x.Name);
+            foreach (var colInfo in Transformer.Columns)
+            {
+                if (!inputSchema.TryFindColumn(colInfo.input, out var col))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.input);
+                if (!(col.ItemType is TextType) || col.Kind != SchemaShape.Column.VectorKind.VariableVector)
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.input, Transformer.InputType.ToString(), col.GetTypeString());
+
+                result[colInfo.output] = new SchemaShape.Column(colInfo.output, SchemaShape.Column.VectorKind.Vector, Transformer.OutputType.ItemType, false);
+            }
+
+            return new SchemaShape(result.Values);
+        }
+    }
+
+    public static class WordEmbeddingsStaticExtensions
+    {
+        public static Vector<float> WordEmbeddings(this Vector<string> input, WordEmbeddingsTransform.PretrainedModelKind modelKind = WordEmbeddingsTransform.PretrainedModelKind.Sswe,
```

Comment:
>Vector [](start = 56, length = 6)

I believe this must be over `VarVector`, not `Vector`, is it not so? Indeed given the structure of your test, where you do call this as far as I see, I'm not sure how that test is succeeding, since I imagine it would have failed?

Reply:
It because the output tokens vector (`SentimentText_Features_TransformedText`) produced by the TextFeatrizer is of type `Vector` currently. I set it `VarVector` and it fails there actually.

---
In reply to: [218171746](https://github.com/dotnet/machinelearning/pull/928#discussion_r218171746) [](ancestors = 218171746)

Reply:
Ah. That sounds like a mistake. The transformed tokenized text will definitely be `VarVector<string>` not `Vector<string>`. We ought to fix that there, and in any event not propagate the error there.

---
In reply to: [218183133](https://github.com/dotnet/machinelearning/pull/928#discussion_r218183133) [](ancestors = 218183133,218171746)

Reply:
I think @Zruty0 might be able to comment more on this.

---
In reply to: [218219010](https://github.com/dotnet/machinelearning/pull/928#discussion_r218219010) [](ancestors = 218219010,218183133,218171746)

Reply:
ok, I am pushing a new commit without fixing it and wil  talk to @Zruty0 for more detail.

---
In reply to: [218219076](https://github.com/dotnet/machinelearning/pull/928#discussion_r218219076) [](ancestors = 218219076,218219010,218183133,218171746)

Reply:
Fixed by getting a VarVector in AssertStatic.

var data2 = dynamicData.AssertStatic(Env, ctx => (
                SentimentText_Features_TransformedText: ctx.Text.VarVector,
                SentimentText: ctx.Text.Scalar,
                label: ctx.Bool.Scalar));

Is this is the correct behavior for this method that it does not complain about getting Vector type when underlying type is VarVector?

---
In reply to: [218223351](https://github.com/dotnet/machinelearning/pull/928#discussion_r218223351) [](ancestors = 218223351,218219076,218219010,218183133,218171746)

Reply:
Hi Zeeshan, I think `SentimentText_Features_TransformedText` that you're asserting from *is* a variable sized vector, so asserting that it is a `VarVector` as you have done here seems correct to me. Is it not so?

(On the other hand, we ought to have some mechanism to convert a `Vector` to `VarVector`, since that seems like something that might be useful sometimes.)

---
In reply to: [218239184](https://github.com/dotnet/machinelearning/pull/928#discussion_r218239184) [](ancestors = 218239184,218223351,218219076,218219010,218183133,218171746)